### ISSUE TITLE
Incoherent Kernel Selector

### ISSUE THREAD

**Issue by echarles** [jupyterlab/jupyterlab#14692]
Here are 2 cases where the Notebook kernel selector (upper-right) show incoherent behavior. For the following examples, I have 4 kernels (a, b, z and python3), all are python language, and I start the server with `--MultiKernelManager.default_kernel_name="b"` to indicate the default kernel.

For comparison, this what is shown when you select form the menu File > New > Notebook - This may sound logical as the preferred kernel `b` is preselect and alone listed in the default section

<img width="484" alt="issue1_a" src="https://github.com/jupyterlab/jupyterlab/assets/226720/b18a8b0d-80be-49e6-b612-c866a6e0aabc">

## Case 1 - Open an Notebook assigned to an existing KernelSpec

The following 2 pictures are showing the kernel selector for Notebooks assigned to existing KernelSpec. This is not coherent with the selector shown when opening a Notebook from the menu. Indeed the list of preferred kernels is populated with all kernelspecs with language `python`, so all `python` kernels are listed as potential default, the effectively assigned kernels is preselected.

<img width="471" alt="issue1_b" src="https://github.com/jupyterlab/jupyterlab/assets/226720/25a3459b-2273-4ede-9033-f1afa6f448ee">

<img width="458" alt="issue1_c" src="https://github.com/jupyterlab/jupyterlab/assets/226720/1cef4267-2714-4e16-99e6-abcf8d16f238">

## Case 2 - Open an Notebook assigned to an non existing KernelSpec

In that case, the all python kernels are listed, and the first one is preselected. The user should be shown with the selector shown when opening a Notebook from the menu.

<img width="490" alt="Screenshot 2023-06-15 at 15 06 23" src="https://github.com/jupyterlab/jupyterlab/assets/226720/56c6f544-d40d-4baa-9d21-fb173f5352c6">

## Case 3 - Open a Notebook not assigned to a Kernel

The following is shown which is coherent.

<img width="480" alt="Screenshot 2023-06-15 at 15 11 39" src="https://github.com/jupyterlab/jupyterlab/assets/226720/99b499cc-869b-4213-9287-683c7da8394e">

**Comment by echarles** [comment]
I have just opened https://github.com/jupyterlab/jupyterlab/pull/14693 which adresses case 2 incoherence (the most critical).

About case 1, I realise that "preferred" may be interpreted in different ways:

- The system administrator defined with `MultiKernelManager.default_kernel_name` trait
- The kernel defined in the notebook metadata.

For now, I see the implementation does not make distinction between those two notions. I can adapt the code, but would need advice/review for a UX expert to define what wording and categories need to be shown to the user in the different cases.

**Comment by echarles** [comment]
In case 1, we could a new category  `Current Kernel` on top of the existing ones which will contain the current kernel. WDYT?

**Comment by echarles** [comment]
> Hi @echarles, so in the 1st case what is the expected behavior? I didn't quite understand what should be selected in that case.

The selected kernel was ok, but the grouping and the group content were IMHO not coherent.

> Regarding the other cases and the overall wording of this list, I am working on a new information architecture and improved wording, because I agree it is very confusing. I might open a separate issue to discuss it.

Thx.

**Comment by GabrielaVives** [comment]
> The selected kernel was ok, but the grouping and the group content were IMHO not coherent.
> 
> >

Do you have an idea of a better grouping?

**Comment by echarles** [comment]
> Do you have an idea of a better grouping?

I have ideas but does not find them very good. The call for a UX designer was to start from a pure user perspective rather than from a developer one.

**Comment by GabrielaVives** [comment]
Could you explain your ideas? It might give others some ideas as well. I am not a user so my input won't be very valuable until I understand what makes sense for you (I suppose you do use this feature? ).

**Comment by GabrielaVives** [comment]
I created the issue #14717 to collect feedback from users regarding the information architecture of this drop-down.

**Comment by JasonWeill** [comment]
Blocked on #14717, to inform us about the intended design.
